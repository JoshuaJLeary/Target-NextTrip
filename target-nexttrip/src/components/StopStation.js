import React, { Component } from 'react';
import { connect } from 'react-redux';

class StopStation extends Component {
  // handles input value for nextTripInput.stopStation
  handleStopStationChange = (event) => {
    const { onChange, dispatch } = this.props;
    onChange({ stopStation: event.target.value });
    setTimeout( () => {
      dispatch({ type: 'GET_ROUTE_DEPARTURE', payload: this.props.nextTripInput.nextTrip })
    }, 500);
  }

  render() {
    // loop through nextTripRouteStop and return each index as an option element
    let routeOptions = this.props.nextTripRouteStop.map(( routeStop ) => {
      return (
        <option key={routeStop.Value} value={routeStop.Value}>
          {routeStop.Text}
        </option>
      )
    })

    return (
      <div data-testid={'stopStation'} className='inputSelectContainer'>
        <select className='inputSelect' onChange={this.handleStopStationChange}>
          <option>Select Stop/Station</option>
          {routeOptions}
        </select>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  state
})

export default connect(mapStateToProps)(StopStation);
