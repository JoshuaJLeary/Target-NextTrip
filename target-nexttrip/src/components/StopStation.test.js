import React from 'react';
import ReactDOM from 'react-dom';
import StopStation from './StopStation';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { createStore, applyMiddleware } from 'redux';
import reducer from '../reducers/reduxStore';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';

const mockStopStation = [
  {
    Text: 'Target Field Station Platform 1',
    Value: 'TF12'
  }
];

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  (applyMiddleware(sagaMiddleware)),
);

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <StopStation nextTripRouteStop={mockStopStation}/>
    </Provider>, div);
});

it('renders StopStation correctly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <StopStation nextTripRouteStop={mockStopStation}/>
    </Provider>);
  expect(getByTestId('stopStation')).toHaveTextContent('Target Field Station Platform 1');
 });

 it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <StopStation nextTripRouteStop={mockStopStation}/>
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
