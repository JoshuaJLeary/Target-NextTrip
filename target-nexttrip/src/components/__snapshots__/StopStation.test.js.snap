// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="inputSelectContainer"
  data-testid="stopStation"
>
  <select
    className="inputSelect"
    onChange={[Function]}
  >
    <option>
      Select Stop/Station
    </option>
    <option
      value="TF12"
    >
      Target Field Station Platform 1
    </option>
  </select>
</div>
`;
